src/cascadia/TerminalApp/CascadiaSettings.cpp

```diff
@@ -643,3 +643,128 @@ Profile CascadiaSettings::_CreateDefaultProfile(const std::wstring_view name)
+        // be able to gracefully use the default settings.
+
+        // We can't add the warning to the list of warnings here, because this
+        // object is not going to be returned at any point.
+
+        throw ::TerminalApp::TerminalException(::TerminalApp::SettingsLoadErrors::NoProfiles);
+    }
+}
+
+// Method Description:
+// - Checks if the "globals.defaultProfile" is set to one of the profiles we
+//   actually have. If the value is unset, or the value is set to something that
+//   doesn't exist in the list of profiles, we'll arbitrarily pick the first
+//   profile to use temporarily as the default.
+// - Appends a SettingsLoadWarnings::MissingDefaultProfile to our list of
+//   warnings if we failed to find the default.
+void CascadiaSettings::_ValidateDefaultProfileExists()
+{
+    const auto defaultProfileGuid = GlobalSettings().GetDefaultProfile();
+    const bool nullDefaultProfile = defaultProfileGuid == GUID{};
+    bool defaultProfileNotInProfiles = true;
+    for (const auto& profile : _profiles)
+    {
+        if (profile.GetGuid() == defaultProfileGuid)
+        {
+            defaultProfileNotInProfiles = false;
+            break;
+        }
+    }
+
+    if (nullDefaultProfile || defaultProfileNotInProfiles)
+    {
+        _warnings.push_back(::TerminalApp::SettingsLoadWarnings::MissingDefaultProfile);
+        // Use the first profile as the new default
+
+        // _temporarily_ set the default profile to the first profile. Because
+        // we're adding a warning, this settings change won't be re-serialized.
+        GlobalSettings().SetDefaultProfile(_profiles[0].GetGuid());
+    }
+}
+
+// Method Description:
+// - Checks to make sure there aren't any duplicate profiles in the list of
+//   profiles. If so, we'll remove the subsequent entries (temporarily), as they
+//   won't be accessible anyways.
+// - Appends a SettingsLoadWarnings::DuplicateProfile to our list of warnings if
+//   we find any such duplicate.
+void CascadiaSettings::_ValidateNoDuplicateProfiles()
+{
+    bool foundDupe = false;
+
+    std::vector<size_t> indiciesToDelete{};
+
+    // Helper to establish an ordering on guids
+    struct GuidEquality
+    {
+        bool operator()(const GUID& lhs, const GUID& rhs) const
+        {
+            return memcmp(&lhs, &rhs, sizeof(rhs)) < 0;
+        }
+    };
+    std::set<GUID, GuidEquality> uniqueGuids{};
+
+    // Try collecting all the unique guids. If we ever encounter a guid that's
+    // already in the set, then we need to delete that profile.
+    for (int i = 0; i < _profiles.size(); i++)
+    {
+        if (!uniqueGuids.insert(_profiles.at(i).GetGuid()).second)
+        {
+            foundDupe = true;
+            indiciesToDelete.push_back(i);
+        }
+    }
+
+    // Remove all the duplicates we've marked
+    // Walk backwards, so we don't accidentally shift any of the elements
+    for (int reverse = indiciesToDelete.size() - 1; reverse >= 0; reverse--)

```

Comment:
Uhh, I feel you could do this loop with a reverse iterator of `indicesToDelete` instead of using an integer count. (like using `.rbegin()` and `.rend()` as the boundary conditions).